crates/runtime/src/dataconnector/snowflake.rs

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use super::DataConnector;
+use super::DataConnectorFactory;
+use async_trait::async_trait;
+use data_components::snowflake::SnowflakeTableFactory;
+use data_components::Read;
+
+use datafusion::datasource::TableProvider;
+use db_connection_pool::snowflakepool::SnowflakeConnectionPool;
+use db_connection_pool::DbConnectionPool;
+use secrets::Secret;
+use snafu::prelude::*;
+use snowflake_api::SnowflakeApi;
+use spicepod::component::dataset::Dataset;
+use std::any::Any;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::{collections::HashMap, future::Future};
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Unable to create flight client: {source}"))]
+    UnableToCreateFlightClient { source: flight_client::Error },
+
+    #[snafu(display("{source}"))]
```

Comment:
Unable to get read provider: {source}

Reply:
Propose to currently keep as-is as this is what we do for all data connectors to make errors more user friendly. I'll test this and update if needed. Thank you.

Reply:
With a little bit more info like other display will help debug too. I think the user friendly error may happen on another interception layer.